### PR TITLE
[AIRFLOW-5048] Improve display of Kubernetes resources

### PR DESCRIPTION
**Pull request by ms32035**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira]
  - https://issues.apache.org/jira/browse/AIRFLOW-5048

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

At the moment just a serialized class is displayed in task instance view. This PR improves string conversion

### Tests

- [X] My PR adds the following unit tests: 

`test_display_resources`

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.

### Code Quality

- [X] Passes `flake8`

### CODE REVIEW

File: airflow/kubernetes/pod.py
Comment (andriisoldatenko) [inline airflow/kubernetes/pod.py]: `self.request_memory` can be None, may be check first?
same for `self.limit_memory`

Comment (ms32035) [reply airflow/kubernetes/pod.py]: true, the test covers this, `None` to string is `'None'`

**Review by potiuk — COMMENTED** [review]
We've just merged GPU limits as well so you should add them.

**Comment by potiuk** [comment]
Tests fail on python 3.5

**Comment by potiuk** [comment]
I think now they will fail on python 3.6 I am afraid. You should use OrderedDict if you want to maintain order @ms32035

**Comment by ms32035** [comment]
I'll go for a simpler solution instead of trying to have dict ordering solution compatible with both 3.5 and 3.6

**Comment by ms32035** [comment]
@potiuk current test failures are not related to the change

**Comment by potiuk** [comment]
@ms32035 -> please rebase to master (but only after we merge https://github.com/apache/airflow/pull/5703 as it fixes broken master due to dependency problem)

**Review by dimberman — APPROVED** [review]
@potiuk this all LGTM but I'll hold off unless you have any other concerns.

**Review by potiuk — APPROVED** [review]

**Pull request merged**
